### PR TITLE
fix: no-non-null-assertion

### PR DESCRIPTION
**Pull request by sdstolworthy**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by danielleadams** [comment]
Is this ready for review?

File: packages/amplify-appsync-simulator/API.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ export class AmplifyAppSyncSimulator {
     // (undocumented)
     start(): Promise<void>;
     // (undocumented)
-    stop(): void;
+    stop(): Promise<void>;
```
Comment (evcodes) [inline packages/amplify-appsync-simulator/API.md:98]: We need to make sure that this is not a breaking change with the data team. I believe we need to reach out and get alignment on this.

File: packages/amplify-appsync-simulator/src/index.ts [hunk 2]
```diff
@@ -123,8 +123,8 @@ export class AmplifyAppSyncSimulator {
     await this._server.start();
   }
 
-  stop() {
-    this._server.stop();
+  async stop() {
+    await this._server.stop();
```
Comment (evcodes) [inline packages/amplify-appsync-simulator/src/index.ts:127]: same here

**Comment by danielleadams** [comment]
@sdstolworthy are you still working on this?

**Comment by sdstolworthy** [comment]
@danielleadams No, closing it out. This needs to be broken into more manageable PRs.

**Pull request closed without merging**